internal/engine/k8swatch/pod_watch.go

```diff
@@ -160,11 +191,22 @@ func (w *PodWatcher) setupNewUIDs(ctx context.Context, st store.RStore, newUIDs
 	w.mu.Lock()
 	defer w.mu.Unlock()
 
+	state := st.RLockState()
```

Comment:
seems generally bad to unlock the state multiple times in a dispatch, because you could get inconsistent state between calls. can you copy out the state you need into the taskList?